hbase_sql/src/main/java/com/splicemachine/olap/OlapServerSubmitter.java

```diff
@@ -390,6 +409,22 @@ private String prepareCommands(String exec, String parameters) {
                 }
             }
         }
+        // If user does not specify a kerberos keytab or principal, use HBase master's.
+        if (UserGroupInformation.isSecurityEnabled()) {
+            Configuration configuration = HConfiguration.unwrapDelegate();
+            String principal = System.getProperty(PRINCIPAL_KEY);
+            String keytab = System.getProperty(KEYTAB_KEY);
+            if (principal == null || keytab == null) {
+                principal = configuration.get(HBASE_MASTER_PRINCIPAL_KEY);
+                String hostname = InetAddress.getLocalHost().getHostName();
+                SpliceLogUtils.warn(LOG, "Trying to get local hostname. This could be problem for host with multiple interfaces.");
```

Comment:
Do we want to add some checks to ensure that the hostnames appear with multiple interfaces, if yes, print this warning (should reduce the warning noise when possible). What happens currently if we proceed with multiple interfaces, errors out or still works?

Reply:
It will use wrong principal and error out

Reply:
We should use NetworkUtils.getHostname() here, it uses hbase.master.hostname if specified rather than relying on InetAddress.getLocalHost().getHostName(), which won't work properly with multiple interfaces